src/core_plugins/kibana/public/management/sections/indices/add_data_steps/parse_csv_step/lib/__tests__/validate_headers.js

```diff
@@ -0,0 +1,83 @@
+import validateHeaders from '../validate_headers';
+import expect from 'expect.js';
+
+describe('validateHeaders', function () {
+
+  describe('argument validations', function () {
+
+    it('should throw an error if the first argument is not an array', () => {
+      expect(validateHeaders).withArgs('foo').to.throwError();
+      expect(validateHeaders).withArgs(1234).to.throwError();
+      expect(validateHeaders).withArgs(undefined).to.throwError();
+      expect(validateHeaders).withArgs(null).to.throwError();
+      expect(validateHeaders).withArgs([]).to.not.throwError();
+    });
+  });
+
+  describe('simple validations', function () {
+
+    it('should return empty array if fields are valid', function () {
+      const errors = validateHeaders(['aa', 'bb', 'cc', 'dd', 'ee']);
+      expect(errors).to.eql([]);
+    });
+
+    it('should return empty array if it contains white space field name', function () {
+      const errors = validateHeaders(['aa', 'bb', '  ', ' ', 'ee']);
+      expect(errors.length).to.be(0);
+    });
+
+    it('should return an error if contains blank field name', function () {
+      const errors = validateHeaders(['aa', 'bb', 'cc', '', 'ee']);
+      expect(errors.length).to.be(1);
+      expect(errors[0]).eql({ type: 'blank', positions: [4] });
+    });
+
+    it('should return an error if fields contain duplicate field name', function () {
+      const errors = validateHeaders(['aa', 'bb', 'cc', 'dd', 'aa']);
+      expect(errors.length).to.be(1);
+      expect(errors[0]).eql({ type: 'duplicate', positions: [1, 5], fieldName: 'aa' });
+    });
+
+    it('should return multiple errors if fields contain duplicate and blank field name', function () {
+      const errors = validateHeaders(['aa', 'bb', '', 'dd', 'aa']);
+      expect(errors.length).to.be(2);
+      expect(errors[0]).eql({ type: 'duplicate', positions: [1, 5], fieldName: 'aa' });
+      expect(errors[1]).eql({ type: 'blank', positions: [3] });
+    });
+  });
+
+  describe('validations with 1000 columns', function () {
```

Comment:
the tests are super clear, love  it!

IMO, I would remove this 1000-column test. It's a lot of pomp and circumstance to test for multiple duplicated field names and blanks. I'd add another short&sweet 4-line stub like the ones above just for that case.